### PR TITLE
MISUV-7392 Enter new POA amount page WIP

### PR DESCRIPTION
**Pull request by danielmaltz1**

### CODE REVIEW

File: app/controllers/claimToAdjustPoa/EnterPoAAmountController.scala [hunk 1]
```diff
@@ -16,32 +16,99 @@
...
+            showInternalServerError(isAgent)
+        }
     }
 
+  def submit(isAgent: Boolean): Action[AnyContent] = auth.authenticatedAction(isAgent) {
+    implicit request =>
+      ifAdjustPoaIsEnabled(isAgent) {
+        claimToAdjustService.getEnterPoAAmountViewModel(Nino(request.nino)).flatMap {
+          case Right(viewModel) =>
+            EnterPoaAmountForm.checkValueConstraints(EnterPoaAmountForm.form.bindFromRequest(), viewModel.totalAmountOne, viewModel.relevantAmountOne).fold(
```
Comment (avelx) [inline app/controllers/claimToAdjustPoa/EnterPoAAmountController.scala:74]: might worth to move whole EnterPoaAmountForm.checkValueConstraints .. section under private method, tidy refactoring to improve readability

File: test/controllers/claimToAdjustPoa/WhatYouNeedToKnowControllerSpec.scala [hunk 2]
```diff
@@ -96,6 +78,23 @@ class WhatYouNeedToKnowControllerSpec extends MockAuthenticationPredicate
         status(resultAgent) shouldBe OK
       }
     }
+    "redirect to the home page" when {
+      "FS is disabled" in {
+        disable(AdjustPaymentsOnAccount)
+        println("BOOP" + isEnabled(AdjustPaymentsOnAccount))
```
Comment (avelx) [inline test/controllers/claimToAdjustPoa/WhatYouNeedToKnowControllerSpec.scala:84]: remove println / debug code pls

File: app/utils/ClaimToAdjustUtils.scala [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import auth.MtdItUser
+import config.featureswitch.FeatureSwitching
+import controllers.routes
+import models.admin.AdjustPaymentsOnAccount
+import play.api.mvc.Result
+import play.api.mvc.Results.Redirect
+
+import scala.concurrent.Future
+
```
Comment (avelx) [inline app/utils/ClaimToAdjustUtils.scala:27]: :thinking: - as I undertand our intention for ClaimToAdjust/POA work would be to use this trait across most of the controllers?

File: app/forms/adjustPoa/EnterPoaAmountForm.scala [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+  private val emptyError = "claimToAdjustPoa.enterPoaAmount.emptyError"
+  private val sameError = "claimToAdjustPoa.enterPoaAmount.sameError"
+  private val higherError = "claimToAdjustPoa.enterPoaAmount.higherError"
+  private val invalidError = "claimToAdjustPoa.enterPoaAmount.invalidError"
+
+  val form: Form[EnterPoaAmountForm] = Form(
+    mapping(
+      amount -> of(currencyFormatter())
+        //bigDecimal.verifying(firstError(isValidNumber))
```
Comment (avelx) [inline app/forms/adjustPoa/EnterPoaAmountForm.scala:43]: you can drop commented code here

File: app/services/ClaimToAdjustService.scala [hunk 5]
```diff
@@ -65,6 +62,38 @@ class ClaimToAdjustService @Inject()(val financialDetailsConnector: FinancialDet
...
+          case Right(None) => Right(None)
+          case Left(ex) => Left(ex)
+        }
+      case Right(None) => Future.successful(Right(None))
+      case Left(ex) => Future.successful(Left(ex))
+    }
+  }
+
+  def getEnterPoAAmountViewModel(nino: Nino)(implicit hc: HeaderCarrier, user: MtdItUser[_], ec: ExecutionContext): Future[Either[Throwable, PoAAmountViewModel]] = {
+    for {
```
Comment (avelx) [inline app/services/ClaimToAdjustService.scala:79]: line 80 and 81 would call getPoaForNonCrystallisedFinancialDetails;
essentially this is duplicate call to the API we should avoid, if you can leave TODO to address this ? in the worst case we can move it into tech debt

File: app/utils/ClaimToAdjustUtils.scala (hunk 3 above)
Comment (danielmaltz1) [reply app/utils/ClaimToAdjustUtils.scala:27]: Yes Pavel, I've implemented this one the other POA pages as well. It cuts down on duplicated code. We had a similar thing with WithIncomeSourceFS

**Review by avelx — APPROVED** [review]

**Pull request merged**